pkg/auth/image_proxy.go

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2022. ByteDance Ltd. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package auth
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/containerd/containerd/defaults"
+	"github.com/containerd/containerd/log"
+	"github.com/containerd/containerd/pkg/dialer"
+	"github.com/containerd/containerd/reference"
+	distribution "github.com/containerd/containerd/reference/docker"
+	"github.com/containerd/stargz-snapshotter/service/keychain/cri"
+	"github.com/containerd/stargz-snapshotter/service/resolver"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/backoff"
+	"google.golang.org/grpc/credentials/insecure"
+	runtime "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+)
+
+const defaultImageServiceAddress = "/run/containerd/containerd.sock"
+
+var Cred resolver.Credential
+
+// from stargz-snapshotter/cmd/containerd-stargz-grpc/main.go#main
+func AddImageProxy(ctx context.Context, rpc *grpc.Server, imageServiceAddress string) {
+	criAddr := defaultImageServiceAddress
+	if ia := imageServiceAddress; ia != "" {
+		criAddr = ia
+	}
```

Comment:
This if block looks a little strange. Why do we need an intermediate `ia` rather than directly assigning imageServiceAddress to criAddr

Reply:
the source code has been updated but looks like the code here is not refreshed 🤣